includes/REST_API/Stories_Controller.php

```diff
@@ -132,6 +133,35 @@ public function prepare_item_for_response( $post, $request ): WP_REST_Response {
 			$data['embed_post_link'] = add_query_arg( [ 'from-web-story' => $post->ID ], admin_url( 'post-new.php' ) );
 		}
 
+		/*
+		 * Internally, WordPress uses a special post_date_gmt value of 0000-00-00 00:00:00
+		 * to indicate that a draft's date is "floating" and should be updated whenever the post is saved.
+		 * This makes it much more difficult for API clients to know the correct date of a draft post.
+		 *
+		 * @link https://core.trac.wordpress.org/ticket/38883
+		 * @link https://github.com/GoogleForCreators/web-stories-wp/issues/11403
+		 */
+		if ( 'auto-draft' === $post->post_status || 'draft' === $post->post_status ) {
```

Comment:
Looks like doing this for `draft` as well breaks the list view in the dashboard, which expects a proper date for drafts.